### ISSUE TITLE
Feature request: Add auth to /universe/structures/ endpoint

### ISSUE THREAD

**Issue by ETristram** [ccpgames/esi-issues#560]
# Feature request

### Route
/universe/structures/

### Authentication
esi-universe.read_structures
esi-markts.structure_markets

### Use case
Adam4EVE.eu provides statistics on structures in New Eden as well as the markets out there.
In order to do this, all structures IDs from the public endpoint /universe/structures/ are pulled and afterwards these IDs are run through both authed endpoints /markets/structures/{structure_id}/  and /universe/structures/{structure_id}/ .

Currently there is AFAIK no way to determine before the query to the authed endpoints, if a 4xx return code "Access forbidden" will be returned. Thus a lot of these responses happen, if all of the structures are queried in order to check access and fetch data.

These "Access forbidden" responses also count against the planned Error Rate Limiting and thus if parallel querying is used after 100 "Forbidden" responses a X-ESI-Error-Limited ban would occur.

One could introduce a pause or throttle the queries in order to avoid the ban but that would defeat the intention of the Error Rate Limiting to reduce the number of 4xx or 5xx request responses.
Therefore a way to determine in advance if the two authed structure endpoint will return data would be needed.

My suggestion is to extend the existing /universe/structures/ with authentication to return bools if structure details and market data can be retrieved with the given character.

### Example returns
{
  "application/json": {
    "id": 1000000017013,
    "details": 1,
    "market": 0
  }
}

**Comment by a-tal** [comment]
this sounds similar to #254

**Comment by ETristram** [comment]
True, with the added aspect of the structure details endpoint and the now imminent impact of the Error Rate Limiting

**Comment by ETristram** [comment]
Dear Team Tech Co., with Error Rate Limiting going live on Monday, is there any best practice for the issue described here? Other than pausing before the X-Esi-Error-Limited hits?

**Comment by Aidansavage** [comment]
It may be worth collating information about the structures (alliance/corp ownership, etc). As long as you arent pulling 100 structures owned by the same corp/alliance at the same time, in the event the first structure owned by corp/alliance XYZ returns a 403 or other error regarding access, you would be able to skip over all the rest owned by XYZ. This might help reduce errors enough to slide past the error limits.

**Comment by ETristram** [comment]
And from which unauthed endpoint do you get the owner?
Furthermore the ACL is per structure and not corp-wide, isn't it? Thus the same owner could have a freeport and a private structure or am I wrong here?

**Comment by krojew** [comment]
Looks like A4E and Evernus are both screwed...

**Comment by ETristram** [comment]
I implemented a workaround waiting for the window to pass if remaining acceptable errors go below 10. Output from todays script run, no ban:

WARN: Need to sleep for 2s to avoid ban!
WARN: Need to sleep for 3s to avoid ban!
WARN: Need to sleep for 3s to avoid ban!

**Comment by krojew** [comment]
And I thought rate-limiting died with CREST... Well, let's see what they say. I really don't want to add waiting, since I'm pushing thousands upon thousands or requests per second.
